macros/tests/basic.rs

```diff
@@ -48,6 +48,31 @@ macro_rules! testcases {
     };
 }
 
+macro_rules! negative_testcases {
+    ($($name:ident { $($grammar:tt)* }: $rule:ident($input:expr) => $expected:expr),*) => {
+        $(mod $name {
+            ::gll_macros::scannerless_parser!($($grammar)*);
+        }
+        #[test]
+        fn $name() {
+            $name::$rule::parse_with($input, |parser, result| {
+                let result = format!("{:#?}", result.unwrap_err());
```

Comment:
I'd much rather prefer if you changed the original to `format!("{:#?}", result)` - then, we can have `Ok(...)` and `Err(...)` in tests.

Reply:
... how did I miss this when you said it. I'm blaming GitHub that I didn't see this until now when I looked up this PR for an unrelated reason 😅

Reply:
Side note: it'd probably make sense to migrate these tests to using (inline) [insta](https://crates.io/crates/insta) tests.